webapp/components/survey/NodeDefDetails/BasicProps.js

```diff
@@ -206,14 +188,11 @@ const BasicProps = (props) => {
 }
 
 BasicProps.propTypes = {
-  validation: PropTypes.object.isRequired,
-  nodeDefKeyEditDisabled: PropTypes.bool.isRequired,
-  nodeDefMultipleEditDisabled: PropTypes.bool.isRequired,
-  editingNodeDefFromDesigner: PropTypes.bool.isRequired,
-
-  setNodeDefParentUuid: PropTypes.func.isRequired,
-  setNodeDefProp: PropTypes.func.isRequired,
-  setNodeDefLayoutProp: PropTypes.func.isRequired,
+  nodeDefState: PropTypes.object.isRequired,
+  setNodeDefState: PropTypes.func.isRequired,
+  keyEditDisabled: PropTypes.bool.isRequired,
+  multipleEditDisabled: PropTypes.bool.isRequired,
+  editingFromDesigner: PropTypes.bool.isRequired,
```

Comment:
are these props used only here? if yes move them in its own state

Reply:
right